### PR TITLE
DOCS-733: improve extra params type link

### PR DESCRIPTION
**Pull request by sguequierre**
* for extra params of type `map[string]interface{}`
* replace go docs struct link with more informative link to go docs blog on maps
* realized this would be an improvement when working on DOCS-214 (document how to use extra params) 
* also replaces arm docommand link to resource base go docs to component-specific go docs link

### CODE REVIEW

File: docs/services/motion/_index.md [hunk 1]
```diff
@@ -155,7 +155,7 @@ moved = await motion.move(component_name=my_gripper, destination=PoseInFrame(ref
 - `constraints` ([Constraints](https://pkg.go.dev/go.viam.com/api/service/motion/v1#Constraints)): Pass in optional [motion constraints](./constraints/).
   By default, motion is unconstrained with the exception of obstacle avoidance.
 
-- `extra` [(map[string]interface{})](https://pkg.go.dev/google.golang.org/protobuf/types/known/structpb): A generic struct, containing extra options to pass to the underlying RPC call.
+- `extra` [(map[string]interface{})](https://go.dev/blog/maps): A generic struct, containing extra options to pass to the underlying RPC call.
```
Comment (sguequierre) [inline docs/services/motion/_index.md:158]:
```suggestion
- `extra` [(map[string]interface{})](https://go.dev/blog/maps): Extra options to pass to the underlying RPC call.
```

File: docs/services/motion/_index.md [hunk 2]
```diff
@@ -263,7 +263,7 @@ As of April 21, 2023, [arm](/components/arm/) is the only component so supported
 
 - `worldState` ([WorldState](https://pkg.go.dev/go.viam.com/rdk/referenceframe#WorldState)): Not used. See [note above](#move-vs-movetoposition).
 
-- `extra` [(map[string]interface{})](https://pkg.go.dev/google.golang.org/protobuf/types/known/structpb): A generic struct, containing extra options to pass to the underlying RPC call.
+- `extra` [(map[string]interface{})](https://go.dev/blog/maps): A generic struct, containing extra options to pass to the underlying RPC call.
```
Comment (sguequierre) [inline docs/services/motion/_index.md:266]:
```suggestion
- `extra` [(map[string]interface{})](https://go.dev/blog/maps): Extra options to pass to the underlying RPC call.
```

File: docs/services/motion/_index.md [hunk 3]
```diff
@@ -389,7 +389,7 @@ gripperPoseInObjectFrame = await motion.get_pose(
   This new frame system builds off the robot's frame system and incorporates the `Transform`s provided.
   If the result of adding the `Transform`s results in a disconnected frame system, an error is thrown.
 
-- `extra` [(map[string]interface{})](https://pkg.go.dev/google.golang.org/protobuf/types/known/structpb): A generic struct, containing extra options to pass to the underlying RPC call.
+- `extra` [(map[string]interface{})](https://go.dev/blog/maps): A generic struct, containing extra options to pass to the underlying RPC call.
```
Comment (sguequierre) [inline docs/services/motion/_index.md:392]:
```suggestion
- `extra` [(map[string]interface{})](https://go.dev/blog/maps): Extra options to pass to the underlying RPC call.
```

**Comment by sguequierre** [comment]
could also use https://pkg.go.dev/golang.org/x/exp/maps (less descriptive) or another alternative?

**Review by npentrel — APPROVED** [review]

**Comment by npentrel** [comment]
I think generally more common to link to the docs but the blog looks more approachable so I think that's a good idea

**Pull request merged**